## Methods

TODO outline sections


### Data
Our analyses used bulk RNA-seq data downloaded from the Recount3 Compendium [CITE] on TODO date.
Before filtering, the dataset contained 317,258 samples of 63,856 genes.

We then preprocessed the data.
To filter out single-cell data, we removed all samples with a sparsity greater than 75 percent.
We also removed all samples marked 'scrna-seq' by recount3's pattern matching method (stored in the metadata as 'recount_pred.pattern.predict.type').

To ensure the samples were comparable, we converted the data to transcripts per million [CITE] using gene lengths from BioMart [CITE].
To ensure the genes' magnitudes were comparable, we performed standardization to scale each gene's range from zero to one.
We kept the 5,000 most variable genes within the dataset.

Samples were labeled with their corresponding tissues using the 'recount_pred.curated.tissue' field in the recount3 metadata.
These labels were based on TODO
A total of TODO samples in our dataset had corresponding tissue labels.

Samples were also labeled with their corresponding sex using labels from Flynn et al. [CITE].
These labels were derived using pattern matching on metadata from the European Nucleotide Archive [CITE].
A total of TODO samples in our dataset had sex labels.

### Data splitting
- Studywise splitting (how is it done and why)
- Samplewise splitting (and cite some papers using samplewise?)

### Models used
- One sentence description of deep and three layer nets
- One sentence description of pytorchLR, sklearnLR, and why I used both

### Model training
- Early stopping, dropout (is this true for 3 layer?), loss, optimizers
- Loss weighting
- Gradient clipping
- Link to model configs dir for hyperparameters
- sklearn used all data, everything else used minibatch optimization
- Ensuring determinism in model training (setting seeds, pytorch magic)

### Binary classification?
Workflow

### Multitissue classification
- Mention this formulation is used for both the tissue and sex prediction
- Which tissues and why
- How is the data split with respect to the tissues
- Metric for performance

### Transfer learning
- Pretraining setup (train on different studies with same labels)
- Data splitting details

### Pretraining
- Pretraining setup (imputation)
- How much data is used for pretraining?
- Matching initializations for pretrained and non-pretrained networks

### Visualizing expression compendium
- Extracted 1000 expression samples at random
- Plotted via UMAP and PCA

### Model logging
- Neptune

### Hardware requirements
OS version, CPU/GPU model, RAM, vRAM, time to run (wall)
