## Results 

### Approach 
We compared the performance of linear and nonlinear models across multiple datasets and tasks (fig. @fig:workflow top).
Our datasets consisted of gene expression and tissue labels from GTEx [@doi:10.1038/ng.2653], expression from Recount3 [@doi:10.1186/s13059-021-02533-6] with tissue labels from the Recount3 metadata and sex labels from Flynn et al. [@doi:10.1186/s12859-021-04070-2], and simulated data.
Before use in model training, we removed scRNA samples, TPM normalized, and zero-one standardized the data.
To avoid leakage between cross-validation folds, we place entire studies into single folds instead of splitting them across folds (fig. @fig:workflow bottom).
We then ran the models on increasingly large training sets to determine how model performance is affected by the amount of training data.

![
Schematic of the model analysis workflow. We evaluate three models on multiple classification problems in three datasets (top). We use studywise splitting by default and evaluate the effects of samplewise splitting and pretraining.
](./images/workflow.svg "Workflow diagram"){#fig:workflow}


### Linear and nonlinear models have similar performance in many tasks
We selected expression data from GTEx [@doi:10.1038/ng.2653] to determine whether linear and nonlinear models performed similarly, as it is a relatively well understood dataset with consistent experimental design across samples.
We first trained our models to differentiate between tissue types on pairs of the five most common tissues in the dataset.
Likely due to the clean nature of the data, all models were able to perform perfectly on these binary classification tasks (fig. @fig:prediction_combined top right).

Because the binary classification task was too easy to determine any difference between models, evaluated the models on a more challenging task.
Namely, we tested the models on their ability to perform multiclass classification on all 31 tissues present in the dataset.
The multitask setting showed logistic regression slightly outperforming the five-layer neural network, which in turn slightly outperformed the three-layer net (fig. @fig:prediction_combined top left).

We then validated our findings in a separate dataset: Sequence Read Archive [@doi:10.1093/nar/gkq1019] samples from the Recount3 [@doi:10.1186/s13059-021-02533-6] dataset.
Again we compared the models' ability to differentiate between pairs of tissues (supp. fig. @fig:recount-binary) and found their performance was roughly equivalent.
We also evaluated the models' performance on a multiclass classification problem differentiating between the 21 most common tissues in the dataset.
As in the GTEx setting, the logistic regression model outperformed the five-layer network, which outperformed the three-layer network (fig. @fig:prediction_combined bottom left).

To examine whether these results held in a problem domain other than tissue type prediction, we used our models to predict metadata-derived sex labels (fig. @fig:prediction_combined bottom right), a task previously studied by Flynn et al. [@doi:10.1186/s12859-021-04070-2].
We used the same experimental setup as in our other binary prediction tasks to train the models, but rather than using tissue labels we used metadata-derived sex labels.
In this setting we found that while the models all performed similarly, the nonlinear models tended to have a slight edge over the linear one.

![
Performance of models across four classification tasks. In each panel the loess curve and its 95% confidence interval are plotted based on points from three seeds, ten data subsets, and five folds of studywise cross-validation (for a total of 150 points per model per panel).
](./images/full_signal_combined.svg ){#fig:prediction_combined}

### There is predictive nonlinear signal in biological problems
One can imagine a world where all the signal relevant to tissue prediction is linear.
If that were the case, nonlinear models like neural networks would fail to give any advantage in a prediction problem.
To determine whether there is nonlinear signal in our tissue prediction tasks learnable by our neural nets, we used Limma [@doi:10.1093/nar/gkv007] to remove the linear signal associated with each tissue.

We began by simulating three datasets to better understand model performance for a variety of linear or nonlinear data generating processes.
Our initial dataset simulated both linear and nonlinear signal by generating two types of features: half of the features with a linear dividing line between the simulated classes and half with a nonlinear dividing line (see [Methods](#methods) for more detail).
After training to classify the simulated dataset, all models effectively predicted the simulated classes.
After removing the linear signal from the dataset, nonlinear models remained able to predict the correct classes, but logistic regression was no better than random (fig @fig:simulation middle).

To measure the models' performance in data with only linear signal, we generated another simulated dataset consisting solely of features with a linear dividing line between the classes.
As before, all models were able to predict the different classes well.
However, once the linear signal was removed, all models had accuracy no better than random guessing (fig @fig:simulation left).
That the nonlinear models only achieved baseline accuracy also indicated that the signal removal method was not injecting nonlinear signal into data where nonlinear signal did not exist.

We also trained the models on a dataset where all features were Gaussian noise as a negative control.
As expected, the models all performed at baseline accuracy both before and after the signal removal process (fig. @fig:simulation right).
This finding supported our decision to perform signal removal on the training and validation sets separately, as removing the signal in the full dataset may introduce predictive signal into this setting (supp. fig. @fig:split-signal-correction).

![
Performance of models in binary classification of simulated data before and after signal removal. Dotted lines indicate expected performance for a naive baseline classifier that predicts the most frequent class.
](./images/simulated_data_combined.svg ){#fig:simulation}

When we ran our models on the signal-removed data from GTEx and Recount3, we found that the neural nets performed better than the baseline while logistic regression did not (fig. @fig:signal_removed top right, supp. fig. @fig:recount-binary-combined).
Similarly, the multiclass problems had the logistic regression model performing poorly, while the nonlinear models had performance that increased with an increase in data while remaining worse than before the linear signal was removed (fig. @fig:signal_removed left).
Likewise, the sex label prediction task showed a marked difference between the neural networks and logistic regression: only the neural networks could learn from the data (fig. @fig:signal_removed bottom right).
In each of the settings, the models performed less well than when run on data without signal, indicating an increase in the problem's difficulty, and logistic regression, in particular, performed no better than random.

![
Performance of models across four classification tasks before and after signal removal
](./images/signal_removed_combined.svg ){#fig:signal_removed}

To verify that our results were not an artifact of how we assigned samples to cross-validation folds, we compared the method we used to assign folds with an alternate method called samplewise splitting.
Samplewise splitting (see [Methods](#methods)) is common in machine learning but leaks information between the train and validation sets when used in transcriptomic tasks.
To avoid this data leakage, we split the dataset at the study level when that information was available.
We found that there is, in fact, a significant degree of performance inflation evident when comparing the samplewise split results to the studywise split results in the Recount3 multiclass setting (supp. fig. @fig:splitting).
While this supports our decision to use studywise splitting, the relative performance of each model stays the same regardless of the data splitting technique.

Another common usage pattern in machine learning is training models on a general dataset and fine-tuning them on a dataset of interest.
To ensure that our results were not made irrelevant by different behavior in the pretraining context, we examined the performance of the models with and without pretraining (supp. fig @fig:pretrain).
To do so, we split the Recount3 data into three sets: pretraining, training, and validation (fig. @fig:workflow bottom), then trained two identically initialized copies of each model.
One copy was trained solely on the training data, while the other was trained on the pretraining data and fine-tuned on the training data.
The pretrained models showed high performance even when trained with small amounts of data from the training set.
However, the nonlinear models did not have a greater performance gain from pretraining than logistic regression, and the balanced accuracy was similar across models.
